google/cloud/spanner/internal/connection_impl_test.cc

```diff
@@ -1290,21 +1292,78 @@ TEST(ConnectionImplTest, CommitGetSession_Retry) {
             EXPECT_EQ(db.FullName(), request.database());
             return MakeSessionsResponse({"test-session-name"});
           });
+  EXPECT_CALL(*mock, BeginTransaction(_, _)).WillOnce(Return(txn));
   EXPECT_CALL(*mock, Commit(_, _))
-      .WillOnce([](grpc::ClientContext&,
-                   spanner_proto::CommitRequest const& request) {
+      .WillOnce([&txn](grpc::ClientContext&,
+                       spanner_proto::CommitRequest const& request) {
         EXPECT_EQ("test-session-name", request.session());
-        EXPECT_TRUE(request.has_single_use_transaction());
+        EXPECT_EQ(txn.id(), request.transaction_id());
         return Status(StatusCode::kPermissionDenied, "uh-oh in Commit");
       });
   auto commit = conn->Commit({MakeReadWriteTransaction(), {}});
   EXPECT_EQ(StatusCode::kPermissionDenied, commit.status().code());
   EXPECT_THAT(commit.status().message(), HasSubstr("uh-oh in Commit"));
 }
 
+TEST(ConnectionImplTest, CommitBeginTransaction_Retry) {
```

Comment:
Nit: Similar comment to before about underscores in test names.

Reply:
Filed https://github.com/googleapis/google-cloud-cpp-spanner/issues/1144